src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -414,67 +414,89 @@ private async Task OnPartialValidationSucceededAsync(ChainedHeader chainedHeader
         {
             this.logger.LogTrace("({0}:'{1}')", nameof(chainedHeader), chainedHeader);
 
-            List<ChainedHeaderBlock> chainedHeaderBlocksToValidate;
-            ConnectBlocksResult connectBlocksResult = null;
+            bool shouldPartialValidationContinue = false;
+            ConnectBlocksResult fullResult = null;
+            PartialValidationSucceededResult partialResult = null;
 
             using (await this.reorgLock.LockAsync().ConfigureAwait(false))
             {
-                bool fullValidationRequired;
-
                 lock (this.peerLock)
                 {
-                    chainedHeaderBlocksToValidate = this.chainedHeaderTree.PartialValidationSucceeded(chainedHeader, out fullValidationRequired);
+                    partialResult = this.chainedHeaderTree.PartialValidationSucceeded(chainedHeader);
+                    if (partialResult.ChainedHeaderBlocksToValidate == null)
+                    {
+                        this.logger.LogTrace("(-)[NO_FURTHER_BLOCKS_TO_VALIDATE]");
+                        return;
+                    }
+                    else
+                        shouldPartialValidationContinue = true;
                 }
 
-                this.logger.LogTrace("Full validation is{0} required.", fullValidationRequired ? "" : " NOT");
+                this.logger.LogTrace("Full validation is{0} required.", partialResult.FullValidationRequired ? string.Empty : " NOT");
 
-                if (fullValidationRequired)
+                if (partialResult.FullValidationRequired)
                 {
-                    connectBlocksResult = await this.FullyValidateLockedAsync(chainedHeader).ConfigureAwait(false);
+                    fullResult = await this.FullyValidateAsync(chainedHeader).ConfigureAwait(false);
+                    shouldPartialValidationContinue = fullResult.Succeeded;
```

Comment:
Do you need this boolean? `shouldPartialValidationContinue ` the result is anyway available in `fullResult.Succeeded`

Reply:
We only check the full validation result if it is required. Please see the comment a little bit further down which explains the logic from @aprogiena.

"So, if there was no full validation, partial validation should continue. And if there was full validation, partial validation should continue only if full validation succeeded."